File: Components/WinForms/CS/EditorsMainDemo/Modules/LookUpProperties.cs
using System;
using System.Collections;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Windows.Forms;
using DevExpress.XtraEditors.Repository;
using DevExpress.XtraEditors;
// <icbHeaderClickMode>
// <icbSearchMode>
using DevExpress.XtraEditors.Controls;
// </icbSearchMode>
// </icbHeaderClickMode>

namespace DevExpress.XtraEditors.Demos {
    public partial class LookUpProperties : UserControl {
        public LookUpProperties(LookUpEdit lookUp) {
            InitializeComponent();
            InitComboBoxes();
            InitLookUp(lookUp);
        }

        RepositoryItemLookUpEdit lookUpProperties;
        bool updateValues = false;

        public void InitLookUp(LookUpEdit lookUp) {
            this.lookUpProperties = lookUp.Properties;
            updateValues = true;
            seColumnIndex.Properties.MaxValue = lookUpProperties.Columns.Count - 1;
            seColumnIndex.Enabled = seColumnIndex.Properties.MaxValue > 0;
            lbDisplay.Text = lookUpProperties.DisplayMember;
            lbValue.Text = lookUpProperties.ValueMember;
            seColumnIndex.Value = lookUpProperties.AutoSearchColumnIndex;
            icbSearchMode.EditValue = lookUpProperties.SearchMode;
            icbHeaderClickMode.EditValue = lookUpProperties.HeaderClickMode;
            ceCaseSensitiveSearch.Checked = lookUpProperties.CaseSensitiveSearch;
            seItemHeight.Value = lookUpProperties.DropDownItemHeight;
            teNullText.Text = lookUpProperties.NullText;
            ceHotTrack.Checked = lookUpProperties.HotTrackItems;
            ceShowFooter.Checked = lookUpProperties.ShowFooter;
            ceShowHeader.Checked = lookUpProperties.ShowHeader;
            ceShowLines.Checked = lookUpProperties.ShowLines;
            updateValues = false;
        }
        void InitComboBoxes() {
            // <icbHeaderClickMode>
            icbHeaderClickMode.Properties.Items.AddEnum(typeof(HeaderClickMode));
            // </icbHeaderClickMode>
            // <icbSearchMode>
            icbSearchMode.Properties.Items.AddEnum(typeof(SearchMode));
            // </icbSearchMode>
        }

        private void LookUpProperties_Load(object sender, System.EventArgs e) {
        }

        // <seColumnIndex>
        private void seColumnIndex_EditValueChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.AutoSearchColumnIndex = Convert.ToInt32(seColumnIndex.Value);
        }
        // </seColumnIndex>

        // <icbHeaderClickMode>
        private void icbHeaderClickMode_SelectedIndexChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.HeaderClickMode = (HeaderClickMode)icbHeaderClickMode.EditValue;
        }
        // </icbHeaderClickMode>

        // <icbSearchMode>
        private void icbSearchMode_SelectedIndexChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.SearchMode = (SearchMode)icbSearchMode.EditValue;
        }
        // </icbSearchMode>

        // <ceCaseSensitiveSearch>
        private void ceCaseSensitiveSearch_CheckedChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.CaseSensitiveSearch = ceCaseSensitiveSearch.Checked;
        }
        // </ceCaseSensitiveSearch>

        // <seItemHeight>
        private void seItemHeight_EditValueChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.DropDownItemHeight = Convert.ToInt32(seItemHeight.Value);
        }
        // </seItemHeight>

        // <teNullText>
        private void teNullText_EditValueChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.NullText = teNullText.Text;
        }
        // </teNullText>

        // <ceHotTrack>
        private void ceHotTrack_CheckedChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.HotTrackItems = ceHotTrack.Checked;
        }
        // </ceHotTrack>

        // <ceShowFooter>
        private void ceShowFooter_CheckedChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.ShowFooter = ceShowFooter.Checked;
        }
        // </ceShowFooter>

        // <ceShowHeader>
        private void ceShowHeader_CheckedChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.ShowHeader = ceShowHeader.Checked;
        }
        // </ceShowHeader>

        // <ceShowLines>
        private void ceShowLines_CheckedChanged(object sender, System.EventArgs e) {
            // <skip>
            if(updateValues) return;
            // </skip>
            lookUpProperties.ShowLines = ceShowLines.Checked;
        }
        // </ceShowLines>
    }
}

File: Components/WinForms/VB/EditorsMainDemo/Modules/LookUpProperties.vb
Imports System.Collections
Imports System.ComponentModel
Imports DevExpress.XtraEditors.Repository
Imports DevExpress.XtraEditors
' <icbHeaderClickMode>
' <icbSearchMode>
Imports DevExpress.XtraEditors.Controls
' </icbSearchMode>
' </icbHeaderClickMode>

Namespace DevExpress.XtraEditors.Demos
	Partial Public Class LookUpProperties
		Inherits UserControl

		Public Sub New(ByVal lookUp As LookUpEdit)
			InitializeComponent()
			InitComboBoxes()
			InitLookUp(lookUp)
		End Sub

		Private lookUpProperties As RepositoryItemLookUpEdit
		Private updateValues As Boolean = False

		Public Sub InitLookUp(ByVal lookUp As LookUpEdit)
			Me.lookUpProperties = lookUp.Properties
			updateValues = True
			seColumnIndex.Properties.MaxValue = lookUpProperties.Columns.Count - 1
			seColumnIndex.Enabled = seColumnIndex.Properties.MaxValue > 0
			lbDisplay.Text = lookUpProperties.DisplayMember
			lbValue.Text = lookUpProperties.ValueMember
			seColumnIndex.Value = lookUpProperties.AutoSearchColumnIndex
			icbSearchMode.EditValue = lookUpProperties.SearchMode
			icbHeaderClickMode.EditValue = lookUpProperties.HeaderClickMode
			ceCaseSensitiveSearch.Checked = lookUpProperties.CaseSensitiveSearch
			seItemHeight.Value = lookUpProperties.DropDownItemHeight
			teNullText.Text = lookUpProperties.NullText
			ceHotTrack.Checked = lookUpProperties.HotTrackItems
			ceShowFooter.Checked = lookUpProperties.ShowFooter
			ceShowHeader.Checked = lookUpProperties.ShowHeader
			ceShowLines.Checked = lookUpProperties.ShowLines
			updateValues = False
		End Sub
		Private Sub InitComboBoxes()
			' <icbHeaderClickMode>
			icbHeaderClickMode.Properties.Items.AddEnum(GetType(HeaderClickMode))
			' </icbHeaderClickMode>
			' <icbSearchMode>
			icbSearchMode.Properties.Items.AddEnum(GetType(SearchMode))
			' </icbSearchMode>
		End Sub

		Private Sub LookUpProperties_Load(ByVal sender As Object, ByVal e As System.EventArgs) Handles MyBase.Load
		End Sub

		' <seColumnIndex>
		Private Sub seColumnIndex_EditValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles seColumnIndex.EditValueChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.AutoSearchColumnIndex = Convert.ToInt32(seColumnIndex.Value)
		End Sub
		' </seColumnIndex>

		' <icbHeaderClickMode>
		Private Sub icbHeaderClickMode_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles icbHeaderClickMode.SelectedIndexChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.HeaderClickMode = CType(icbHeaderClickMode.EditValue, HeaderClickMode)
		End Sub
		' </icbHeaderClickMode>

		' <icbSearchMode>
		Private Sub icbSearchMode_SelectedIndexChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles icbSearchMode.SelectedIndexChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.SearchMode = CType(icbSearchMode.EditValue, SearchMode)
		End Sub
		' </icbSearchMode>

		' <ceCaseSensitiveSearch>
		Private Sub ceCaseSensitiveSearch_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ceCaseSensitiveSearch.CheckedChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.CaseSensitiveSearch = ceCaseSensitiveSearch.Checked
		End Sub
		' </ceCaseSensitiveSearch>

		' <seItemHeight>
		Private Sub seItemHeight_EditValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles seItemHeight.EditValueChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.DropDownItemHeight = Convert.ToInt32(seItemHeight.Value)
		End Sub
		' </seItemHeight>

		' <teNullText>
		Private Sub teNullText_EditValueChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles teNullText.EditValueChanged
			' <skip>
            If updateValues Or lookUpProperties Is Nothing Then
                Return
            End If
			' </skip>
			lookUpProperties.NullText = teNullText.Text
		End Sub
		' </teNullText>

		' <ceHotTrack>
		Private Sub ceHotTrack_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ceHotTrack.CheckedChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.HotTrackItems = ceHotTrack.Checked
		End Sub
		' </ceHotTrack>

		' <ceShowFooter>
		Private Sub ceShowFooter_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ceShowFooter.CheckedChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.ShowFooter = ceShowFooter.Checked
		End Sub
		' </ceShowFooter>

		' <ceShowHeader>
		Private Sub ceShowHeader_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ceShowHeader.CheckedChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.ShowHeader = ceShowHeader.Checked
		End Sub
		' </ceShowHeader>

		' <ceShowLines>
		Private Sub ceShowLines_CheckedChanged(ByVal sender As Object, ByVal e As System.EventArgs) Handles ceShowLines.CheckedChanged
			' <skip>
			If updateValues Then
				Return
			End If
			' </skip>
			lookUpProperties.ShowLines = ceShowLines.Checked
		End Sub
		' </ceShowLines>
	End Class
End Namespace
